.github/workflows/publishing.yml

```diff
@@ -158,6 +160,49 @@ jobs:
         env:
           GH_TOKEN: ${{ secrets.REPO_BOT_ACCESS_TOKEN }}
 
+      - name: Retrieve wheel assets
+        run: |
+          assets_folder=/tmp/wheel_assets && mkdir "$assets_folder" && cd "$assets_folder"
+          
+          function download_assets {
+            mkdir "$1" && cd "$1"
+            release_id=`echo "$2" | jq -r ".$1.release_id"`
+            echo "Download $release_id assets..."
+            gh release download $release_id -R ${{ github.repository }}
+            cd ..
+          }
+          
+          for file in ${{ join(fromJson(steps.artifacts.outputs.python_wheels).info_files, ' ') }}; do
+            download_assets $file '${{ steps.artifacts.outputs.python_wheels }}'
+          done
+         
+        env:
+          GH_TOKEN: ${{ secrets.REPO_BOT_ACCESS_TOKEN }}
+  
+      - name: Upload wheel assets
+        uses: actions/upload-artifact@v3
+        with:
+          name: wheel_assets
+          path: /tmp/wheel_assets
+          retention-days: 1
+          if-no-files-found: error    
+      
+      # We can detele staging branch now after downloading the assets (as artifacts for later wheel jobs). 
+      - name: Clean up
+        run: |
+          # Clean up the staging branch that was used to trigger the GitLab pipeline.
+          git config --global user.name "cuda-quantum-bot"
+          git config --global user.email "cuda-quantum-bot@users.noreply.github.com"
+          function delete_staging_branch {
+            staging_branch=`echo "$2" | jq -r ".$1.staging_branch"`
+            echo "Delete staging branch $staging_branch ..."
+            git push origin --delete $staging_branch || true
```

Comment:
This looks like it could potentially accidentally delete some branches. (Maybe it doesn't right now, but as people make updates to this file, it could  have unintended side effects. Would it be possible to do any sanity checking of `$staging_branch` before performing this command? For example, something like this bash:
```
if [[ "$staging_branch" =~ ^regex$ ]]; then
  echo "matched (and do the work)"
else
  echo "did not match (and throw an error)"
fi
```

Reply:
Yea, makes sense. I've added a regex check as suggested.